src/config/config.rs

```diff
@@ -112,17 +112,16 @@ impl Config {
 pub struct Type {
   pub fields: BTreeMap<String, Field>,
   pub doc: Option<String>,
-  pub interface: Option<bool>,
-  pub implements: Option<Vec<String>>,
-  #[serde(rename = "enum")]
+  #[serde(default)]
+  pub interface: bool,
+  #[serde(default)]
+  pub implements: Vec<String>,
+  #[serde(rename = "enum", default)]
   pub variants: Option<Vec<String>>,
```

Comment:
Note that I haven't changed this. It is because `variants` are using as discriminated value to separate enums from other types and to remove it we need some other flag/switch to handle it as enum properly